azurerm/internal/services/containers/kubernetes_nodepool.go

```diff
@@ -153,6 +153,11 @@ func SchemaDefaultNodePool() *schema.Schema {
 					ForceNew:     true,
 					ValidateFunc: computeValidate.ProximityPlacementGroupID,
 				},
+				"only_critical_addons_enabled": {
```

Comment:
could we add an acceptance test creating a cluster where this is enabled for the default node pool?

Reply:
Certainly, added, results in the comment.